### DISCUSSION TITLE
How can I obtain screen data from proxy?

### DISCUSSION THREAD

**Discussion by Nukaccino** [FreeRDP/FreeRDP#8793] [Q&A] [answered]
In client program, dumping `gdi->primary_buffer` was enough to generate image since this is decoded bitmap. But how can I do the same in proxy? Exporting image from `pf_client_end_paint` in `pf_update.c` give me only black screen.

**Comment by akallabeth** [comment, marked as answer]
you can´t. Proxy does not decode data but passes it on.

**Reply by Nukaccino** [reply]
I thought I can decode data before pass, but is this totally impossible? I want to get a hint where I should put decoding process If decoding is possible on proxy.

**Reply by akallabeth** [reply]
you connect a second client to it and decode it there. (we already support stream recording to file/pipe/...)

**Reply by Nukaccino** [reply]
Thank you. I'll give a try in that way.
